rust/kernel/net/device.rs

```diff
@@ -0,0 +1,1076 @@
+    BRIDGE_PORT = bindings::netdev_priv_flags_IFF_BRIDGE_PORT,
+    /// Device used as Open vSwitch datapath port.
+    OVS_DATAPATH = bindings::netdev_priv_flags_IFF_OVS_DATAPATH,
+    /// The interface supports sharing skbs on transmit.
+    TX_SKB_SHARING = bindings::netdev_priv_flags_IFF_TX_SKB_SHARING,
+    /// Supports unicast filtering.
+    UNICAST_FLT = bindings::netdev_priv_flags_IFF_UNICAST_FLT,
+    /// Device used as team port.
+    TEAM_PORT = bindings::netdev_priv_flags_IFF_TEAM_PORT,
+    /// Device supports sending custom FCS.
+    SUPP_NOFCS = bindings::netdev_priv_flags_IFF_SUPP_NOFCS,
+    /// Device supports hardware address change when it's running.
+    LIVE_ADDR_CHANGE = bindings::netdev_priv_flags_IFF_LIVE_ADDR_CHANGE,
+    /// Macvlan device.
+    MACVLAN = bindings::netdev_priv_flags_IFF_MACVLAN,
+    /// IFF_XMIT_DST_RELEASE not taking into account underlying stacked devices.
+    XMIT_DST_RELEASE_PERM = bindings::netdev_priv_flags_IFF_XMIT_DST_RELEASE_PERM,
+    /// Device is an L3 master device.
+    L3MDEV_MASTER = bindings::netdev_priv_flags_IFF_L3MDEV_MASTER,
+    /// Device can run without qdisc attached.
+    NO_QUEUE = bindings::netdev_priv_flags_IFF_NO_QUEUE,
+    /// Device is a Open vSwitch master.
+    OPENVSWITCH = bindings::netdev_priv_flags_IFF_OPENVSWITCH,
+    /// Device is enslaved to an L3 master device.
+    L3MDEV_SLAVE = bindings::netdev_priv_flags_IFF_L3MDEV_SLAVE,
+    /// Device is a team device.
+    TEAM = bindings::netdev_priv_flags_IFF_TEAM,
+    /// Device has had Rx Flow indirection table configured.
+    RXFH_CONFIGURED = bindings::netdev_priv_flags_IFF_RXFH_CONFIGURED,
+    /// The headroom value is controlled by an external entity (i.e. the master device for bridged veth).
+    PHONY_HEADROOM = bindings::netdev_priv_flags_IFF_PHONY_HEADROOM,
+    /// Device is a MACsec device.
+    MACSEC = bindings::netdev_priv_flags_IFF_MACSEC,
+    /// Device doesn't support the rx_handler hook.
+    NO_RX_HANDLER = bindings::netdev_priv_flags_IFF_NO_RX_HANDLER,
+    /// Device is a failover master device.
+    FAILOVER = bindings::netdev_priv_flags_IFF_FAILOVER,
+    /// Device is lower dev of a failover master device.
+    FAILOVER_SLAVE = bindings::netdev_priv_flags_IFF_FAILOVER_SLAVE,
+    /// Only invoke the rx handler of L3 master device.
+    L3MDEV_RX_HANDLER = bindings::netdev_priv_flags_IFF_L3MDEV_RX_HANDLER,
+    /// Rename is allowed while device is up and running.
+    LIVE_RENAME_OK = bindings::netdev_priv_flags_IFF_LIVE_RENAME_OK,
+}
+
+/// Feature flags.
+pub mod feature {
+    use core::convert::From;
+    use core::ops::{Add, AddAssign, Deref, DerefMut, Sub, SubAssign};
+
+    /// Holds multiple flags to give to an interface via [`super::NetDevice::add_flag`].
+    #[repr(transparent)]
+    #[derive(Debug, Clone, Copy)]
+    pub struct NetIF(u64);
+
+    impl NetIF {
+        /// Create new Flag with value `0`.
+        pub const fn new() -> Self {
```

Comment:
new -> empty
add_flag -> insert
remove_flag -> remove

Bitflags is the de-facto standard in Rust ecosystem, so it'll be better to keep naming convention identical (we might want to use bitflags in the future).